Linked block decompression (preset dictionaries) broken on Go 1.17

The amd64 assembler version of decodeBlock fails the linked block test with Go 1.17 on my Linux box:

```
$ go test
--- FAIL: TestReader (0.00s)
    --- FAIL: TestReader/testdata/Mark.Twain-Tom.Sawyer_linked.txt.lz4_[] (0.00s)
        reader_test.go:105: lz4: invalid source or destination buffer too short
        reader_test.go:109: invalid size: got 1000; want 4266361
        reader_test.go:113: uncompressed data does not match original
    --- FAIL: TestReader/testdata/Mark.Twain-Tom.Sawyer_linked.txt.lz4_[ConcurrencyOption(8)] (0.00s)
        reader_test.go:105: lz4: invalid source or destination buffer too short
        reader_test.go:109: invalid size: got 5000; want 4266361
        reader_test.go:113: uncompressed data does not match original
FAIL
exit status 1
FAIL	github.com/pierrec/lz4/v4	2.002s
```

The ARM and noasm versions are fine.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -11,7 +11,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
-        go: ["1.14", "1.15"]
+        go: ["1.15", "1.16", "1.17"]
     runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@v2

--- a/internal/lz4block/decode_amd64.s
+++ b/internal/lz4block/decode_amd64.s
@@ -203,6 +203,9 @@ memmove_lit:
 	ADDQ dst_len+8(FP), R8
 	MOVQ src_base+24(FP), R9
 	ADDQ src_len+32(FP), R9
+	MOVQ dict_base+48(FP), R14
+	MOVQ dict_len+56(FP), R15
+	ADDQ R14, R15
 	MOVQ R8, R12
 	SUBQ $32, R12
 	MOVQ R9, R13

